### PR TITLE
fs.watch(mac): hold loop.mutex in FSEvents _events_cb

### PR DESCRIPTION
**Pull request by robobun**
## What

Fix a use-after-free race on macOS where `FSEventsLoop._events_cb` iterated `loop.watchers` on the CoreFoundation thread without holding `loop.mutex`, while the JS thread could concurrently null and destroy an `FSEventsWatcher`.

## Why

`_events_cb` is registered as the `FSEventStream` callback and runs on the dedicated CF thread. It did:

```zig
for (loop.watchers.slice()) |watcher| {            // no lock
    if (watcher) |handle| {
        const handle_path = handle.path;           // UAF
        ...
        handle.emit(event_type.toEvent(path), is_file);
        handle.flush();
    }
}
```

Meanwhile on the JS thread, `watcher.close()` → `FSWatcher.detach()` → `PathWatcher.deinit()` → `FSEventsWatcher.deinit()` does:

```zig
loop.unregisterWatcher(this);   // locks mutex, sets watchers[i] = null, unlocks
bun.default_allocator.destroy(this);
```

…and `PathWatcher.deinit` then frees `resolved_path` (which `handle.path` aliases). Since `_events_cb` never took the mutex, `unregisterWatcher` could return and the watcher could be freed while the CF thread still held a live `handle` pointer and was reading `handle.path` / calling `handle.emit()`.

A concurrent `registerWatcher` could also reallocate the `BabyList` backing buffer while `_events_cb` held a slice into the old one.

`unregisterWatcher` also never rescheduled the `FSEventStream`, so after the last watcher on a path was removed the stream kept firing `_events_cb` for that path indefinitely.

## How

- `_events_cb` now holds `loop.mutex` for the entire iteration, matching `_schedule`. `unregisterWatcher` therefore blocks until the CF thread is done with the watcher, so by the time it returns (and the caller destroys the watcher + path) the CF thread is guaranteed to see `null` on the next pass.
- `unregisterWatcher` now enqueues `_schedule` (same coalescing as `registerWatcher`) so the `FSEventStream` is rebuilt without the removed path instead of firing for it forever.

Lock ordering is safe: the CF thread takes `loop.mutex` → `FSWatcher.mutex` (via `emit` → `refTask`). The JS thread's `close()` releases `FSWatcher.mutex` *before* calling `detach()` → `unregisterWatcher` (which takes `loop.mutex`), so there's no AB-BA cycle.

## Verification

Added `test/js/node/watch/fs.watch.events-cb-race.test.ts` (macOS-only) that floods a watched directory with writes while rapidly creating and closing watchers on it. Under ASAN on an unpatched build the CF thread hits heap-use-after-free; with the fix it completes cleanly.

`zig:check-all` passes on all targets. Existing `fs.watch.test.ts` and `fs.watch.deadlock.test.ts` still pass (the two permission-test failures in `fs.watch.test.ts` are pre-existing on main and happen because the container runs as root).

### CODE REVIEW

File: test/js/node/watch/fs.watch.events-cb-race.test.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    const iterations = 400;
+    let i = 0;
+    function churn() {
+      if (i++ >= iterations) {
+        stopped = true;
+        // Let any in-flight CF callbacks drain before declaring success.
+        setTimeout(() => {
+          console.log("OK " + i + " " + writes);
+          process.exit(0);
+        }, 100);
```
[the remark this answers is not in the document]
Comment (robobun) [reply test/js/node/watch/fs.watch.events-cb-race.test.ts]: Dropped the 100ms drain `setTimeout` → `setImmediate` in 6ebecf6 (all watchers are already closed by that point so the delay added nothing).

Kept the 30s watchdog in the child: it's the same pattern the adjacent `fs.watch.deadlock.test.ts` uses, it's `unref()`'d so it doesn't affect exit, and it gives `i=/writes=` diagnostics if the mutex change ever deadlocks — which the outer test timeout alone wouldn't.
